orderer/common/multichannel/registrar.go

```diff
@@ -372,12 +374,34 @@ func (r *Registrar) CreateBundle(channelID string, config *cb.Config) (channelco
 	return channelconfig.NewBundle(channelID, config, r.bccsp)
 }
 
+// ChannelList returns a slice of ChannelInfoShort containing all application channels (excluding the system
+// channel), and ChannelInfoShort of the system channel (nil if does not exist).
+// The URL fields are empty, and are to be completed by the caller.
 func (r *Registrar) ChannelList() types.ChannelList {
-	//TODO
-	return types.ChannelList{
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+
+	list := types.ChannelList{
 		SystemChannel: nil,
 		Channels:      nil,
 	}
+
+	if len(r.chains) == 0 {
+		return list
+	}
+
+	if r.systemChannelID != "" {
+		list.SystemChannel = &types.ChannelInfoShort{Name: r.systemChannelID, URL: ""}
```

Comment:
You really don't need to populate an empty URL

Reply:
Correct